plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHostActivityChecker.java

```diff
@@ -213,6 +270,17 @@ protected boolean verifyActivityOfStorageOnHost(HashMap<StoragePool, List<Volume
         return poolVolMap;
     }
 
+    private boolean isHostServedByNfsPool(Host agent) {
+        List<StoragePoolHostVO> storagesOnHost = storagePoolHostDao.listByHostId(agent.getId());
+        for (StoragePoolHostVO storagePoolHostRef : storagesOnHost) {
+            StoragePoolVO storagePool = this.storagePool.findById(storagePoolHostRef.getPoolId());
+            if (NFS_POOL_TYPE.contains(storagePool.getPoolType())) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
We could use `stream().anyMatch` here.

Reply:
@GutoVeronezi here I think that it is a bit trickier than a normal loop interation with a simple if.
Such as the other one:
```
for (StoragePoolVO pool : zonePools) {
        if (pool.getPoolType() == StoragePoolType.NetworkFilesystem) {
                return true;
        }
}
```

Here the list does not store the object that is being used in the conditional, but instead just an object (`StoragePoolHostVO`) that will serve to link to the object in quesiton (`StoragePoolVO`). Thus the stream .anyMatch would not serve here.

Please correct me if I am wrong in this one.